lib/extension-es6.js

```diff
@@ -7,7 +7,8 @@ function es6(loader) {
 
   var transpiler, transpilerModule, transpilerRuntimeModule, transpilerRuntimeGlobal;
 
-  var isBrowser = typeof window != 'undefined';
+  var isBrowser = typeof window !== 'undefined' && !!window.Window && window instanceof window.Window;
+  var isWorker = typeof self !== 'undefined' && !!self.WorkerGlobalScope && self instanceof WorkerGlobalScope;
```

Comment:
It's common enough to just check existence of `window` that I think that is fine. It's used in a lot of places, so I'm not sure I can see justification for updating it everywhere, and I'd rather be consistent.

For the Worker - yes it seems we do an instanceof. Again consistency would be good here, but I don't see why we need three conditions and not just one actually:

```javascript
var autoloadTranspiler = typeof window !== 'undefined' || typeof WorkerGlobalScope !== 'undefined' && typeof self !== 'undefined';
```